core/src/main/kotlin/supplier/CacheEntitySupplier.kt

```diff
@@ -157,27 +147,28 @@ public class CacheEntitySupplier(private val kord: Kord) : EntitySupplier {
         return Message(data, kord)
     }
 
-    override fun getMessagesAfter(messageId: Snowflake, channelId: Snowflake, limit: Int): Flow<Message> {
-        require(limit > 0) { "At least 1 item should be requested, but got $limit." }
+    override fun getMessagesAfter(messageId: Snowflake, channelId: Snowflake, limit: Int?): Flow<Message> {
+        checkLimit(limit)
         return cache.query<MessageData> {
             idEq(MessageData::channelId, channelId)
             MessageData::id gt messageId
-        }.asFlow().map { Message(it, kord) }.take(limit)
+        }.asFlow().map { Message(it, kord) }.limit(limit)
     }
 
-    override fun getMessagesBefore(messageId: Snowflake, channelId: Snowflake, limit: Int): Flow<Message> {
-        require(limit > 0) { "At least 1 item should be requested, but got $limit." }
+    override fun getMessagesBefore(messageId: Snowflake, channelId: Snowflake, limit: Int?): Flow<Message> {
+        checkLimit(limit)
         return cache.query<MessageData> {
             idEq(MessageData::channelId, channelId)
-            idGt(MessageData::id, messageId)
```

Comment:
make an equvilant function for Lt, it'd be better to have common conversion OVA.